tests/purs/failing/OverlapAcrossModulesUnnamedInstance.out

```diff
@@ -0,0 +1,24 @@
+Error found:
+in module [33mOverlapAcrossModules[0m
+at tests/purs/failing/OverlapAcrossModulesUnnamedInstance.purs:6:1 - 6:15 (line 6, column 1 - line 6, column 15)
+
+  Overlapping type class instances found for
+  [33m                                [0m
+  [33m  OverlapAcrossModules.Class.C X[0m
+  [33m                               Y[0m
+  [33m                                [0m
+  The following instances were found:
+
+    OverlapAcrossModules.X.cX
+    OverlapAcrossModules.$cXY0
```

Comment:
These error messages are not great. Neither of these instances has a programmer-provided name, and while at least the `$` in the second instance suggests this, the first instance has nothing to indicate that the programmer shouldn't go literally searching for an instance named `cX`. But even in the second case, `$cXY0` is not even the final name that will appear in the generated code!

I think the ideal thing here would be to describe these instances with something like this:
```suggestion
    instance in module OverlapAcrossModules.X with type forall y. C X y
    instance in module OverlapAcrossModules with type C X Y
```

I'd like to address this in a later PR.

Reply:
Sounds good!